apollo-router-core/src/spec/query.rs

```diff
@@ -518,21 +497,14 @@ impl Query {
                     skip,
                     include,
                 } => {
-                    if skip
-                        .should_skip(variables)
-                        // validate_variables should have already checked that
-                        // the variable is present and it is of the correct type
-                        .unwrap_or_default()
-                    {
+                    // Using .unwrap_or is legit here because
+                    // validate_variables should have already checked that
+                    // the variable is present and it is of the correct type
```

Comment:
NIT: We probably don't need this comment every time, could we have a block comment at the top of the function which does this?